v2/internal/controllers/to_azure_secrets_test.go

```diff
@@ -66,10 +69,121 @@ func Test_MissingSecretKey_ReturnsError(t *testing.T) {
 
 	tc.CreateResourceAndWaitForState(vm, metav1.ConditionFalse, conditions.ConditionSeverityWarning)
 	// We expect the ready condition to include details of the error
-	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal("SecretNotFound"))
+	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal(conditions.ReasonSecretNotFound))
 	tc.Expect(vm.Status.Conditions[0].Message).To(
 		ContainSubstring("Secret \"%s/%s\" does not contain key \"%s\"", tc.Namespace, secret.Name, secret.Key))
 
 	// Delete VM and resources.
 	tc.DeleteResourcesAndWait(vm, networkInterface, subnet, vnet, rg)
 }
+
+func Test_UserSecretInDifferentNamespace_SecretNotFound(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	secretName := tc.Namer.GenerateName("admin-pw")
+	nameSpaceName := "secret-namespace"
+
+	createNamespaces(tc, nameSpaceName)
+	secretInDiffNamespace := createNamespacedSecret(tc, nameSpaceName, secretName)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	vnet := newVMVirtualNetwork(tc, testcommon.AsOwner(rg))
+	subnet := newVMSubnet(tc, testcommon.AsOwner(vnet))
+	networkInterface := newVMNetworkInterface(tc, testcommon.AsOwner(rg), subnet)
+
+	tc.CreateResourceAndWait(vnet)
+	tc.CreateResourceAndWait(subnet)
+	tc.CreateResourceAndWait(networkInterface)
+
+	vm := newVM(tc, rg, networkInterface, secretInDiffNamespace)
+
+	tc.CreateResourceAndWaitForState(vm, metav1.ConditionFalse, conditions.ConditionSeverityWarning)
+	// We expect the ready condition to include details of the error
+	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal(conditions.ReasonSecretNotFound))
+	tc.Expect(vm.Status.Conditions[0].Message).To(
+		ContainSubstring("failed resolving secret references: %s/%s does not exist", tc.Namespace, secretInDiffNamespace.Name))
+
+	// Delete VM and resources.
+	tc.DeleteResourcesAndWait(vm, networkInterface, subnet, vnet, rg)
+}
+
+func Test_UserSecretInDifferentNamespace_ShouldNotTriggerReconcile(t *testing.T) {
+	t.Parallel()
+	ns1 := "ns-1"
```

Comment:
minor: I think name these namespaces something more closely affiliated with this test... maybe: `secret-in-different-namespace-ns-1` and `secret-in-different-namespace-ns-2`?